fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowProcedureStatusCommand.java

```diff
@@ -59,10 +72,74 @@ public ShowResultSetMetaData getMetaData() {
         return builder.build();
     }
 
+    private String extractFilterValue(PartitionColumnFilter colFilter) {
+
+        InPredicate inPredicate = colFilter.getInPredicate();
+        if (null == inPredicate) {
+            // equal one value
+            if (colFilter.lowerBoundInclusive && colFilter.upperBoundInclusive
+                    && colFilter.lowerBound != null && colFilter.upperBound != null
+                    && 0 == colFilter.lowerBound.compareLiteral(colFilter.upperBound)) {
+                return colFilter.lowerBound.getStringValue();
+            }
+        }
+        throw new AnalysisException("Only supports equalTo in filter for whereClause");
+    }
+
+    private void validateAndExtractFilters(StringBuilder dbFilter, StringBuilder procFilter) throws Exception {
+        Map<String, PartitionColumnFilter> filterMap = Maps.newHashMap();
+        if (!this.whereExpr.isEmpty()) {
+            whereExpr.stream().map(ExpressionUtils::checkAndMaybeCommute).filter(Optional::isPresent)
+                    .forEach(expr -> new ExpressionColumnFilterConverter(filterMap).convert(expr.get()));
+        }
+        // we support filter on Db and Name and ProcedureName.
+        // But one column we can put only once and support conjucts
+        for (Map.Entry<String, PartitionColumnFilter> elem : filterMap.entrySet()) {
+            String columnName = elem.getKey();
+            if ((!columnName.equals("Db"))
+                    && (!columnName.equals("Name"))
+                    && (!columnName.equals("ProcedureName"))) {
+                throw new AnalysisException("Only supports filter Db, Name, ProcedureName with equalTo or LIKE");
```

Comment:
Currently only support "Db", "Name", "ProcedureName" equals and `LIKE`,  which can meet requirements.

Ideally, parse whereClause like query to support any expression filter, is not necessary now.